File: SoftInvModel/src/main/java/pe/edu/pucp/softinv/model/Comentario/ComentarioProductoDTO.java
package pe.edu.pucp.softinv.model.Comentario;

import pe.edu.pucp.softinv.model.Personas.ClienteDTO;
import pe.edu.pucp.softinv.model.Producto.ProductoDTO;

public class ComentarioProductoDTO extends ComentarioDTO{
    private ProductoDTO producto;

    public ProductoDTO getProducto() {
        return producto;
    }

    public void setProducto(ProductoDTO producto) {
        this.producto = producto;
    }

    public ComentarioProductoDTO() {
        super();
        producto = null;
    }

    public ComentarioProductoDTO(Integer idComentario, ClienteDTO cliente, String comentario, Integer valoracion, ProductoDTO producto) {
        super(idComentario,cliente,comentario,valoracion);
        this.producto = producto;
    }
}

File: C#/SoftInvModel/Comentario/comentarioProductoDTO.cs
using System;

namespace softinv.model
{
    public class ComentarioProductoDTO: ComentarioDTO
        {
        private ProductoDTO producto;

        public ProductoDTO Producto { get => producto; set => producto = value; }

        public ComentarioProductoDTO(): base()
        {
            Producto = null;
        }

        public ComentarioProductoDTO(int idComentario, ClienteDTO cliente, String comentario, int valoracion, ProductoDTO producto): base(idComentario, cliente, comentario, valoracion)
        {
            this.Producto = producto;
        }

        public int? IdProducto { get => producto.IdProducto; set => producto.IdProducto = value; }
    }
}
